runtime/gc_include/ObjectAllocationAPI.hpp

```diff
@@ -80,25 +142,30 @@ class MM_ObjectAllocationAPI
 #endif /* J9VM_ENV_DATA64 */
 			{
 				/* Calculate the size of the object */
-				UDATA const headerSize = J9VMTHREAD_CONTIGUOUS_HEADER_SIZE(currentThread);
-				UDATA const dataSize = ((UDATA)size) * J9ARRAYCLASS_GET_STRIDE(arrayClass);
-				UDATA allocateSize = (dataSize + headerSize + _objectAlignmentInBytes - 1) & ~(UDATA)(_objectAlignmentInBytes - 1);
+				uintptr_t const headerSize = J9VMTHREAD_CONTIGUOUS_HEADER_SIZE(currentThread);
+				uintptr_t const dataSize = ((uintptr_t)size) * J9ARRAYCLASS_GET_STRIDE(arrayClass);
+				uintptr_t allocateSize = (dataSize + headerSize + _objectAlignmentInBytes - 1) & ~(uintptr_t)(_objectAlignmentInBytes - 1);
+#if defined(J9VM_ENV_DATA64)
+				if (allocateSize < J9_GC_MINIMUM_INDEXABLE_OBJECT_SIZE) {
+					allocateSize = J9_GC_MINIMUM_INDEXABLE_OBJECT_SIZE;
+				}
+#else /* !J9VM_ENV_DATA64 */
 				if (allocateSize < J9_GC_MINIMUM_OBJECT_SIZE) {
 					allocateSize = J9_GC_MINIMUM_OBJECT_SIZE;
 				}
-
+#endif /* J9VM_ENV_DATA64 */
 				/* Allocate the memory */
 				j9object_t objectHeader = NULL;
 
 				switch(_gcAllocationType) {
 
 #if defined(J9VM_GC_THREAD_LOCAL_HEAP)
 				case OMR_GC_ALLOCATION_TYPE_TLH:
-					if (allocateSize <= ((UDATA) currentThread->heapTop - (UDATA) currentThread->heapAlloc)) {
-						U_8 *heapAlloc = currentThread->heapAlloc;
-						UDATA afterAlloc = (UDATA)heapAlloc + allocateSize;
+					if (allocateSize <= ((uintptr_t) currentThread->heapTop - (uintptr_t) currentThread->heapAlloc)) {
+						uint8_t *heapAlloc = currentThread->heapAlloc;
+						uintptr_t afterAlloc = (uintptr_t)heapAlloc + allocateSize;
```

Comment:
This can be:
```
uint8_t *afterAlloc = heapAlloc + allocateSize;
```
which eliminates the cast here and the need for one on line 168.